src/app/libp2p_helper/src/codanet.go

```diff
@@ -307,40 +330,21 @@ func MakeHelper(ctx context.Context, listenOn []ma.Multiaddr, externalAddr ma.Mu
 		p2p.ConnectionManager(connManager),
 		p2p.ListenAddrs(listenOn...),
 		p2p.AddrsFactory(func(as []ma.Multiaddr) []ma.Multiaddr {
-			if externalAddr == nil {
-				return as
+			if externalAddr != nil {
+				as = append(as, externalAddr) 
 			}
 
-			as = append(as, externalAddr)
-                        
-                        _, exists := os.LookupEnv("CONNECT_PRIVATE_IPS")
-                        if exists { return as }
-                        
-			bs := make([]ma.Multiaddr, 0, len(as))
-			isPrivate := func(addr ma.Multiaddr) bool {
-				// get the ip out
-				// filter against the private ips
-				fs := ma.NewFilters()
-				parseCIDR := func(cidr string) gonet.IPNet {
-					_, ipnet, err := gonet.ParseCIDR(cidr)
-					if err != nil {
-						panic(err)
-					}
-					return *ipnet
-				}
-
-				fs.AddFilter(parseCIDR("10.0.0.0/8"), ma.ActionDeny)
-				fs.AddFilter(parseCIDR("172.16.0.0/12"), ma.ActionDeny)
-				fs.AddFilter(parseCIDR("192.168.0.0/16"), ma.ActionDeny)
-				fs.AddFilter(parseCIDR("100.64.0.0/10"), ma.ActionDeny)
-				fs.AddFilter(parseCIDR("198.18.0.0/15"), ma.ActionDeny)
-				fs.AddFilter(parseCIDR("169.254.0.0/16"), ma.ActionDeny)
+			_, exists := os.LookupEnv("CONNECT_PRIVATE_IPS")
```

Comment:
This logic is wrong. (You didn't introduce this, but we need to fix it, as I think this may be the cause of why we still see private ips shared). I believe the person who wrote this code meant for this to filter out the connection gating logic, not the inclusion of private ips in the addresses we advertise. Can we move this into the connection gating logic as an override to the new private ip filter instead?

Reply:
will do!